### ISSUE TITLE
Multi-Value Deposit Addresses

### ISSUE THREAD

**Issue by digitalchaotics** [knowm/XChange#1816]
The current AccountService API does not accommodate multi-value deposit addresses.  For example on Bittrex, to deposit XMR you are required to provide two values:  the "Payment_id" and the "Base Address".  In another case (BTS) the two values are labeled "Memo" and "Registered Acct".  This issue impacts 18 coins on Bittrex.  I'm not aware of this issue on other exchanges.

This issue impacts the utility of two AccountService API methods:  'withdrawFunds' and 'requestDepositAddress'.  Neither method provides a way to use multi-value addresses.

One solution to this issue is to expand the API to include a DepositAddress class which would be accepted by 'withdrawFunds' and returned by a new 'getDepositAddress' method, e.g. ...

- public DepositAddress getDepositAddress(Currency currency);
- public String withdrawFunds(Currency currency, BigDecimal amount, DepositAddress address);

Suggestions?  Maybe there's a workaround I'm not aware of?  Has anyone else faced this issue?

**Comment by npomfret** [comment]
This is a bit of a work in progress.  I started to add support for _complex_ addresses when I added the support for XRP withdrawals, see `RippleWithdrawFundsParams`.

I don't think there are any other implementations yet.

**Comment by digitalchaotics** [comment]
Yes - the issue seems to be everywhere for XRP, but I've seen additional examples.  Lots of them on Bittrex, but I've also seen it on other exchanges.

**Comment by npomfret** [comment]
I'd be happy to work on this stuff but it won't be for a couple of weeks (new puppy means I don't have time to wash let alone work!)

**Comment by tmmrt** [comment]
is there any update on requestDepositAddress on accountService? I would like to request a usable address that has the depositaddress as well as tag/paymentid for XRP, XMR etc.

**Comment by amjad1210** [comment]
I need this feature myself so will create a PR.
